packages/core/data/mm/world/beneath_the_well.yml

```diff
@@ -29,7 +29,7 @@
   events:
     STICKS: "can_kill_baba_sticks"
     WATER: "has_bottle"
-    RUPEES: "can_fight || has_weapon_range || has_explosives"
+    RUPEES: "(can_fight || has_weapon_range || has_explosives) && soul_keese"
```

Comment:
for what it's worth, i wouldnt consider keese as a valid source of rupees. there's a wallmaster in this section iirc, which are guaranteed rupees. Above one as well should remove Keese.

Reply:
Good catch - changed to Wallmaster here. Apparently Keese drop purple rupees when killed with Light Arrows so I'm adding that as a valid rupees source as well.